tests/test_socket_server.py

```diff
@@ -39,28 +40,23 @@ class SocketServerTestCase(unittest.IsolatedAsyncioTestCase):
     async def asyncSetUp(self):
         self.ctrl = None
         self.writer = None
-        port = 0
         self.mock_ctrl = None
         self.data = None
-        self.srv = SocketServer(port=port, simulation_mode=1)
+        self.srv = SocketServer(port=0, simulation_mode=1)
 
         self.log = logging.getLogger(type(self).__name__)
 
-        self.assertFalse(self.srv._started)
-        asyncio.create_task(self.srv.start())
-        await asyncio.sleep(0.5)
-        self.assertTrue(self.srv._started)
-        # Request the assigned port from the mock controller.
-        port = self.srv.port
-
-        rw_coro = asyncio.open_connection(host="127.0.0.1", port=port)
-        self.reader, self.writer = await asyncio.wait_for(rw_coro, timeout=TIMEOUT)
+        await self.srv.start_task
+        self.assertTrue(self.srv.server.is_serving())
+        self.reader, self.writer = await asyncio.open_connection(
+            host=tcpip.LOCAL_HOST, port=self.srv.port
+        )
 
     async def asyncTearDown(self):
-        if self.srv._started:
-            await self.srv.exit()
+        await self.srv.disconnect()
         if self.writer:
             self.writer.close()
```

Comment:
Please consider adding `await self.writer.wait_closed()`; it *might* eliminate warnings about running tasks when you run in debug mode.